syz-manager/manager.go

```diff
@@ -1004,13 +1061,26 @@ func (mgr *Manager) hubSync() {
 				Minimized: false, // don't trust programs from hub
 			})
 		}
+		reproDropped := 0
+		for _, repro := range r.Repros {
+			_, err := prog.Deserialize(repro)
+			if err != nil {
+				reproDropped++
+				continue
+			}
+			crash := Crash{0, "external repro", nil, repro, true}
+			mgr.mu.Unlock()
+			mgr.hubReproQueue <- &crash
+			mgr.mu.Lock()
+		}
 		mgr.stats["hub add"] += uint64(len(a.Add))
 		mgr.stats["hub del"] += uint64(len(a.Del))
 		mgr.stats["hub drop"] += uint64(dropped)
-		mgr.stats["hub new"] += uint64(len(r.Inputs) - dropped)
-		Logf(0, "hub sync: add %v, del %v, drop %v, new %v, more %v",
-			len(a.Add), len(a.Del), dropped, len(r.Inputs)-dropped, r.More)
-		if len(r.Inputs)+r.More == 0 {
+		mgr.stats["hub new"] += uint64(len(r.Progs) - dropped)
+		mgr.stats["hub repros"] += uint64(len(r.Repros) - reproDropped)
+		Logf(0, "hub sync: send: add %v, del %v, repros: %v; recv: progs: drop %v, new %v, repros: drop: %v, new %v; more %v",
+			len(a.Add), len(a.Del), len(a.Repros), dropped, len(r.Progs)-dropped, reproDropped, len(r.Repros)-reproDropped, r.More)
+		if len(r.Progs)+r.More == 0 {
```

Comment:
Also need to set `a.Repros = nil` below. Otherwise they will be send on each iteration.